pkg/api/api.go

```diff
@@ -44,9 +56,11 @@ type API struct {
 	TweakResponseFunc func(map[string]any)
 	// IsSubPathApi indicates whenever an API is a sub-path API.
 	SubPathAPI bool
-	// Parent is the parent API ID which is requred to download all possible values for sub-path apis.
-	// e.g. key-user-actions that are bound to applications
-	Parent string
+	// Parent is used for SubPath APIs to store information about the configuration type and ID of the related
+	// configuration once Resolved() is called.
+	Parent Config
```

Comment:
I don't like that "sort of dynamic data" inside this struct. However as a first solution i think it is ok.
A possible refactoring would be to provide a an own type for that data that is returned by `Resolve()`consumers of that data would not continue to work with the API directly but have their own data model for API related information

Reply:
That does feel extremely hacky.... Also it changes the concept of "Parent" of the API as in overarching API it's a sub-path of, to being both the parent type and config ID.
Also from what I can see you only ever care about the Type and Dynatrace Object ID - so writing it to a Config object and it's config ID is misusing the Config.
If you want to keep it as dynamic data in the API, I would not reuse the Config but have a dedicated struct limited to the data needed. 

I like the idea to extend Resolve() better